test/e2e/testsuites/azdiskschedulerextender_pod_scheduling_on_failover_multiple_pv_perf_tester.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	v1alpha1ClientSet "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/clientset/versioned/typed/azuredisk/v1alpha1"
+	"sigs.k8s.io/azuredisk-csi-driver/test/e2e/driver"
+)
+
+// AzDiskSchedulerExtenderPodSchedulingOnFailoverMultiplePV will provision required PV(s), PVC(s) and Pod(s)
+// Pod should successfully be re-scheduled on failover/scaling in a cluster with AzDriverNode and AzVolumeAttachment resources
+type AzDiskSchedulerExtenderPodSchedulingOnFailoverMultiplePV struct {
+	CSIDriver              driver.DynamicPVTestDriver
+	AzDiskClientSet        v1alpha1ClientSet.DiskV1alpha1Interface
+	AzNamespace            string
+	Pod                    PodDetails
+	Volume                 VolumeDetails
+	StorageClassParameters map[string]string
+}
+
+func (t *AzDiskSchedulerExtenderPodSchedulingOnFailoverMultiplePV) Run(client clientset.Interface, namespace *v1.Namespace, schedulerName string) {
+	tStatefulSet, cleanup := t.Pod.SetupStatefulset(client, namespace, t.CSIDriver, schedulerName, 4)
+	for i := range cleanup {
+		i := i
+		defer cleanup[i]()
+	}
+
+	// Get the list of available nodes for scheduling the pod
+	nodes := ListNodeNames(client)
+	if len(nodes) < 1 {
+		ginkgo.Skip("need at least 1 nodes to verify the test case. Current node count is %d", len(nodes))
+	}
+
+	ginkgo.By("deploying the statefulset")
+	tStatefulSet.Create()
+
+	ginkgo.By("checking that the pod for statefulset is running")
+	tStatefulSet.WaitForPodReady()
+
+	for i := 1; i <= 4; i++ {
+		for j := 0; j < 4; j++ {
+			go func(podName string) {
+				_ = client.CoreV1().Pods(tStatefulSet.namespace.Name).Delete(context.TODO(), podName, metav1.DeleteOptions{})
```

Comment:
Why not call `DeletePodAndWait` here?

Reply:
I've updated the `DeletePodAndWait` to use it here.